test/cmd/projects.sh

```diff
@@ -35,7 +35,28 @@ os::cmd::expect_success_and_not_text 'oc config view -o jsonpath="{.contexts[*].
 os::cmd::try_until_text 'oc projects' 'test6'
 os::cmd::expect_success_and_text 'oc project test6' 'Now using project "test6"'
 os::cmd::expect_success_and_text 'oc config view -o jsonpath="{.contexts[*].context.namespace}"' '\btest6\b'
-echo 'projects command ok'
 
+# test if namespace is updated when the user tries to edit immutable fields of a project but has rights to edit the namespace
+os::test::junit::declare_suite_start "cmd/projects/namespace_update"
+os::cmd::expect_success_and_text 'oc login -u nstest -p nstest' 'Login successful.' # login standard user
+os::cmd::expect_success_and_text 'oc new-project nstestproj' 'Now using project "nstestproj" on server ' # make project
+# test that standard user cannot edit immutable fields
+os::cmd::expect_failure_and_text 'oc annotate project nstestproj key1=val1 --overwrite' 'The Project "nstestproj" is invalid: metadata.annotations\[key1\]: Invalid value: "val1": field is immutable, try updating the namespace'
+os::cmd::expect_failure_and_text 'oc annotate namespace nstestproj key2=val2 --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+os::cmd::expect_success_and_text 'oc login -u system:admin -n default' 'Login successful.|system:admin' # login as admin
+# test that admin user can edit all fields regardless of which endpoint they go through
+os::cmd::expect_success_and_text 'oc annotate project nstestproj key3=val3 --overwrite' 'project "nstestproj" annotated'
+os::cmd::expect_success_and_text 'oc annotate namespace nstestproj key4=val4 --overwrite' 'namespace "nstestproj" annotated'
+os::cmd::expect_success_and_text 'oc login -u nstest -p nstest' 'Login successful.' # login standard user again
+# test that standard user still cannot edit immutable fields
+os::cmd::expect_failure_and_text 'oc annotate project nstestproj key1=val1 --overwrite' 'The Project "nstestproj" is invalid: metadata.annotations\[key1\]: Invalid value: "val1": field is immutable, try updating the namespace'
+os::cmd::expect_failure_and_text 'oc annotate namespace nstestproj key2=val2 --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+# this will now be successful because there is no diff between the old and new project
```

Comment:
I would be curious to know if you guys consider this behavior valid (seems a bit confusing).